src/main/java/seedu/address/model/util/NextSessionUtil.java

```diff
@@ -0,0 +1,86 @@
+package seedu.address.model.util;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.ResolverStyle;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Session;
+
+/**
+ * Utility class containing helper methods to help with nextSessionFeedback method.
+ */
+public class NextSessionUtil {
+
+    private final Session timeNowSession;
+    private final ArrayList<Session> toSortList;
+    private final HashMap<Session, Person> sessionPersonHashMap;
+
+    /**
+     * Helps ModelManager to add Sessions into toSortList.
+     */
+    private Consumer<Person> toSortListAdder = new Consumer<Person>() {
+        @Override
+        public void accept(Person person) {
+            for (int i = 0; i < person.getSessionList().sessionList.size(); i++) {
+                Session currSession = person.getSessionList().sessionList.get(i);
```

Comment:
would it be better if you created a variable to store `person.getSessionList().sessionList` before the forloop? i think it would be more efficient instead of the program having to repeatedly access the same methods